### PR TITLE
Put labels on weights plot for sasmodels compare app

### PR DESCRIPTION
**Pull request by pkienzle**
There's also code for plotting probability rather than weight, commented out for now but available for future debugging sessions.

### CODE REVIEW

File: sasmodels/weights.py [hunk 1]
```diff
@@ -303,15 +303,21 @@ def plot_weights(model_info, mesh):
...
+        #plt.interactive(True)
+        plt.figure()
         for (_, x, w), s in zip(mesh, labels):
             if len(x) > 1:
-                pylab.plot(x, w, '-o', label=s)
-        pylab.grid(True)
-        pylab.legend()
-        #pylab.show()
+                # For plotting probability rather than weight scale by dx
+                #dx = np.hstack((x[1]-x[0], x[2:] - x[:-2], x[-1] - x[-2]))
```
Comment (caitwolf) [inline sasmodels/weights.py:314]: There looks to be a factor of 2 missing from some of the dx parameters if we are calculating the bins from midpoint to midpoint. The endpoints are are correct as x[1]-x[0] and x[-1]-x[-2] if we are assuming uniform spacing. However, the points in between would be calculated as:
dx[n] = (x[n+1]-x[n])/2 + (x[n]-x[n-1])/2 = (x[n+1]-x[n-1])/2

Suggested code fix:
`dx = np.hstack((x[1]-x[0], (x[2:] - x[:-2])/2, x[-1] - x[-2]))`

**Review by caitwolf — COMMENTED** [review]
The calculation of dx for plotting probability rather than weight in the comments is the only concern I had. Otherwise, everything else looks good when I test the compare app.

**Comment by wpotrzebowski** [comment]
Waiting for @pkienzle to look at

**Review by caitwolf — APPROVED** [review]

**Comment by wpotrzebowski** [comment]
Ready to merge?

**Pull request merged**